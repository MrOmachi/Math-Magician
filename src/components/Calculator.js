import React, { Component } from 'react';
import calculate from './logic/calculator';
import Navbar from './pages/Navbar';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: null,
      operation: null,
    };
  }

  handleClick = (e) => {
    const data = e.target.value;
    const state = calculate(this.state, data);
    this.setState(state);
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <div>
        <Navbar />
        <div className="calc">
          <div className="displayScreen">
            {total}
            {operation}
            {next}
          </div>
          <div className="calcBlock">
            <div className="col1">
              <button
                type="button"
                value="AC"
                onClick={this.handleClick}
                className="btnAc"
              >
                AC
              </button>
              <button
                type="button"
                value="+/-"
                onClick={this.handleClick}
                className="btnAc"
              >
                +/-
              </button>
              <button
                type="button"
                value="%"
                onClick={this.handleClick}
                className="btnAc"
              >
                %
              </button>
              <button
                type="button"
                value="รท"
                onClick={this.handleClick}
                className="btnAc operator"
              >
                /
              </button>
            </div>

            <div className="col1">
              <button
                type="button"
                value={7}
                onClick={this.handleClick}
                className="btnAc"
              >
                7
              </button>
              <button
                type="button"
                value={8}
                onClick={this.handleClick}
                className="btnAc"
              >
                8
              </button>
              <button
                type="button"
                value={9}
                onClick={this.handleClick}
                className="btnAc"
              >
                9
              </button>
              <button
                type="button"
                value="x"
                onClick={this.handleClick}
                className="btnAc operator"
              >
                x
              </button>
            </div>

            <div className="col1">
              <button
                type="button"
                value={4}
                onClick={this.handleClick}
                className="btnAc"
              >
                4
              </button>
              <button
                type="button"
                value={5}
                onClick={this.handleClick}
                className="btnAc"
              >
                5
              </button>
              <button
                type="button"
                value={6}
                onClick={this.handleClick}
                className="btnAc"
              >
                6
              </button>
              <button
                type="button"
                value="-"
                onClick={this.handleClick}
                className="btnAc operator"
              >
                -
              </button>
            </div>

            <div className="col1">
              <button
                type="button"
                value={1}
                onClick={this.handleClick}
                className="btnAc"
              >
                1
              </button>
              <button
                type="button"
                value={2}
                onClick={this.handleClick}
                className="btnAc"
              >
                2
              </button>
              <button
                type="button"
                value={3}
                onClick={this.handleClick}
                className="btnAc"
              >
                3
              </button>
              <button
                type="button"
                value="+"
                onClick={this.handleClick}
                className="btnAc operator"
              >
                +
              </button>
            </div>

            <div className="col1">
              <button
                type="button"
                value={0}
                onClick={this.handleClick}
                className="btnAc btn0"
              >
                0
              </button>
              <button
                type="button"
                value="."
                onClick={this.handleClick}
                className="btnAc"
              >
                .
              </button>
              <button
                type="button"
                value="="
                onClick={this.handleClick}
                className="btnAc operator"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
