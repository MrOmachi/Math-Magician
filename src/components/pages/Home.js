import React from 'react';
import Navbar from './Navbar';

function Home() {
  return (
    <div>
      <Navbar />
      <h2>Welcome to my page</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor ut
        recusandae maiores dignissimos, accusamus laboriosam velit voluptatum.
        Necessitatibus molestiae iure aut excepturi eos numquam, perspiciatis
        laudantium. Porro excepturi doloribus quas odio cupiditate esse, ducimus
        itaque. Odio quam doloribus ea quos dicta modi omnis dolorem culpa,
        provident nisi iste beatae adipisci?
      </p>
    </div>
  );
}

export default Home;
