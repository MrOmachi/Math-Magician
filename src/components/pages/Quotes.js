import React from 'react';
import Navbar from './Navbar';

function Quotes() {
  return (
    <div>
      <Navbar />
      <p>
        Mathematics is not about numbers, equations, computations or algorithms,
        it is about understanding - William Paul Thurston
      </p>
    </div>
  );
}

export default Quotes;
