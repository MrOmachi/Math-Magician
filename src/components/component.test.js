import { render, screen } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import Home from './pages/Home';
import Quotes from './pages/Quotes';
import Calculator from './Calculator';

describe('Testing the App UI', () => {
  test('Testing home UI component', () => {
    const home = render(
      <Router>
        <Home />
      </Router>,
    );
    expect(home).toMatchSnapshot();
  });

  test('Calculator UI component', async () => {
    const calc = render(
      <Router>
        <Calculator />
      </Router>,
    );
    const btn = await screen.findAllByRole('button');
    expect(calc).toMatchSnapshot();
    expect(btn.length).toBe(19);
  });

  test('Quote UI component', () => {
    const quote = render(
      <Router>
        <Quotes />
      </Router>,
    );

    const str = screen.getByText(
      /Mathematics is not about numbers, equations, computations or algorithms, it is about understanding - William Paul Thurston/i,
    );

    expect(quote).toMatchSnapshot();
    expect(str).toBeInTheDocument();
  });
});
