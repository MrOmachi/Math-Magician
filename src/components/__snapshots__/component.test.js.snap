// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App UI Calculator UI component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navigation"
        >
          <header
            style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
          >
            <h1>
              Math Magician
            </h1>
            <ul
              style="display: flex; gap: 20px; list-style: none;"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/Calculator"
                >
                  Calculator
                </a>
              </li>
              <li>
                <a
                  href="/Quotes"
                >
                  Quotes
                </a>
              </li>
            </ul>
          </header>
        </div>
        <div
          class="calc"
        >
          <div
            class="displayScreen"
          />
          <div
            class="calcBlock"
          >
            <div
              class="col1"
            >
              <button
                class="btnAc"
                type="button"
                value="AC"
              >
                AC
              </button>
              <button
                class="btnAc"
                type="button"
                value="+/-"
              >
                +/-
              </button>
              <button
                class="btnAc"
                type="button"
                value="%"
              >
                %
              </button>
              <button
                class="btnAc operator"
                type="button"
                value="รท"
              >
                /
              </button>
            </div>
            <div
              class="col1"
            >
              <button
                class="btnAc"
                type="button"
                value="7"
              >
                7
              </button>
              <button
                class="btnAc"
                type="button"
                value="8"
              >
                8
              </button>
              <button
                class="btnAc"
                type="button"
                value="9"
              >
                9
              </button>
              <button
                class="btnAc operator"
                type="button"
                value="x"
              >
                x
              </button>
            </div>
            <div
              class="col1"
            >
              <button
                class="btnAc"
                type="button"
                value="4"
              >
                4
              </button>
              <button
                class="btnAc"
                type="button"
                value="5"
              >
                5
              </button>
              <button
                class="btnAc"
                type="button"
                value="6"
              >
                6
              </button>
              <button
                class="btnAc operator"
                type="button"
                value="-"
              >
                -
              </button>
            </div>
            <div
              class="col1"
            >
              <button
                class="btnAc"
                type="button"
                value="1"
              >
                1
              </button>
              <button
                class="btnAc"
                type="button"
                value="2"
              >
                2
              </button>
              <button
                class="btnAc"
                type="button"
                value="3"
              >
                3
              </button>
              <button
                class="btnAc operator"
                type="button"
                value="+"
              >
                +
              </button>
            </div>
            <div
              class="col1"
            >
              <button
                class="btnAc btn0"
                type="button"
                value="0"
              >
                0
              </button>
              <button
                class="btnAc"
                type="button"
                value="."
              >
                .
              </button>
              <button
                class="btnAc operator"
                type="button"
                value="="
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navigation"
      >
        <header
          style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
        >
          <h1>
            Math Magician
          </h1>
          <ul
            style="display: flex; gap: 20px; list-style: none;"
          >
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/Calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/Quotes"
              >
                Quotes
              </a>
            </li>
          </ul>
        </header>
      </div>
      <div
        class="calc"
      >
        <div
          class="displayScreen"
        />
        <div
          class="calcBlock"
        >
          <div
            class="col1"
          >
            <button
              class="btnAc"
              type="button"
              value="AC"
            >
              AC
            </button>
            <button
              class="btnAc"
              type="button"
              value="+/-"
            >
              +/-
            </button>
            <button
              class="btnAc"
              type="button"
              value="%"
            >
              %
            </button>
            <button
              class="btnAc operator"
              type="button"
              value="รท"
            >
              /
            </button>
          </div>
          <div
            class="col1"
          >
            <button
              class="btnAc"
              type="button"
              value="7"
            >
              7
            </button>
            <button
              class="btnAc"
              type="button"
              value="8"
            >
              8
            </button>
            <button
              class="btnAc"
              type="button"
              value="9"
            >
              9
            </button>
            <button
              class="btnAc operator"
              type="button"
              value="x"
            >
              x
            </button>
          </div>
          <div
            class="col1"
          >
            <button
              class="btnAc"
              type="button"
              value="4"
            >
              4
            </button>
            <button
              class="btnAc"
              type="button"
              value="5"
            >
              5
            </button>
            <button
              class="btnAc"
              type="button"
              value="6"
            >
              6
            </button>
            <button
              class="btnAc operator"
              type="button"
              value="-"
            >
              -
            </button>
          </div>
          <div
            class="col1"
          >
            <button
              class="btnAc"
              type="button"
              value="1"
            >
              1
            </button>
            <button
              class="btnAc"
              type="button"
              value="2"
            >
              2
            </button>
            <button
              class="btnAc"
              type="button"
              value="3"
            >
              3
            </button>
            <button
              class="btnAc operator"
              type="button"
              value="+"
            >
              +
            </button>
          </div>
          <div
            class="col1"
          >
            <button
              class="btnAc btn0"
              type="button"
              value="0"
            >
              0
            </button>
            <button
              class="btnAc"
              type="button"
              value="."
            >
              .
            </button>
            <button
              class="btnAc operator"
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the App UI Quote UI component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navigation"
        >
          <header
            style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
          >
            <h1>
              Math Magician
            </h1>
            <ul
              style="display: flex; gap: 20px; list-style: none;"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/Calculator"
                >
                  Calculator
                </a>
              </li>
              <li>
                <a
                  href="/Quotes"
                >
                  Quotes
                </a>
              </li>
            </ul>
          </header>
        </div>
        <p>
          Mathematics is not about numbers, equations, computations or algorithms, it is about understanding - William Paul Thurston
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navigation"
      >
        <header
          style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
        >
          <h1>
            Math Magician
          </h1>
          <ul
            style="display: flex; gap: 20px; list-style: none;"
          >
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/Calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/Quotes"
              >
                Quotes
              </a>
            </li>
          </ul>
        </header>
      </div>
      <p>
        Mathematics is not about numbers, equations, computations or algorithms, it is about understanding - William Paul Thurston
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing the App UI Testing home UI component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navigation"
        >
          <header
            style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
          >
            <h1>
              Math Magician
            </h1>
            <ul
              style="display: flex; gap: 20px; list-style: none;"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/Calculator"
                >
                  Calculator
                </a>
              </li>
              <li>
                <a
                  href="/Quotes"
                >
                  Quotes
                </a>
              </li>
            </ul>
          </header>
        </div>
        <h2>
          Welcome to my page
        </h2>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor ut recusandae maiores dignissimos, accusamus laboriosam velit voluptatum. Necessitatibus molestiae iure aut excepturi eos numquam, perspiciatis laudantium. Porro excepturi doloribus quas odio cupiditate esse, ducimus itaque. Odio quam doloribus ea quos dicta modi omnis dolorem culpa, provident nisi iste beatae adipisci?
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navigation"
      >
        <header
          style="display: flex; justify-content: space-between; width: 90%; margin: auto;"
        >
          <h1>
            Math Magician
          </h1>
          <ul
            style="display: flex; gap: 20px; list-style: none;"
          >
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/Calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="/Quotes"
              >
                Quotes
              </a>
            </li>
          </ul>
        </header>
      </div>
      <h2>
        Welcome to my page
      </h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor ut recusandae maiores dignissimos, accusamus laboriosam velit voluptatum. Necessitatibus molestiae iure aut excepturi eos numquam, perspiciatis laudantium. Porro excepturi doloribus quas odio cupiditate esse, ducimus itaque. Odio quam doloribus ea quos dicta modi omnis dolorem culpa, provident nisi iste beatae adipisci?
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
